sink.go

```diff
@@ -57,6 +64,20 @@ func (fh FanoutSink) SetGaugeWithLabels(key []string, val float32, labels []Labe
 	}
 }
 
+func (fh FanoutSink) SetPrecisionGauge(key []string, val float64) {
+	fh.SetPrecisionGaugeWithLabels(key, val, nil)
+}
+
+func (fh FanoutSink) SetPrecisionGaugeWithLabels(key []string, val float64, labels []Label) {
+	for _, s := range fh {
+		if s64, ok := s.(PrecisionGaugeMetricSink); ok {
+			s64.SetPrecisionGaugeWithLabels(key, val, labels)
+		} else {
+			s.SetGaugeWithLabels(key, float32(val), labels)
```

Comment:
> We chose to panic in the general case of a sink not supporting rather than silently downsample.

This wasn't a thought-out decision. Do you think it's worth downsampling the values instead of panicking in case the type check is unsuccessful?

Reply:
Yeah I liked the panic option for straightforward case because I think that makes it harder to accidentally use the library incorrectly if your sink doesn't support the feature.

In this case I'd maybe lean towards doing neither panic or downsample and instead just not call anything on an unsupported sink for a precision gauge. Possibly wrong data seems more dangerous or likely to cause problems than missing data - if the fact it's missing is noticed then it might prompt some research into why whereas it just being wrong is much less likely to be spotted and could lead to incorrect analysis downstream by humans or automation.

Whatever option we should add a doc comment to call it out as a footgun the developer is responsible for dealing with.